dev-build/cmake/cmake-3.30.3.recipe

```diff
@@ -0,0 +1,304 @@
+SUMMARY="A cross-platform build system"
+DESCRIPTION="CMake is a family of tools designed to build, test and package \
+software. It is used to control the software compilation process using \
+simple platform and compiler independent configuration files. CMake generates \
+native makefiles and workspaces that can be used in the compiler environment \
+of your choice."
+HOMEPAGE="https://cmake.org/"
+COPYRIGHT="2002-2024 Kitware, Inc., Insight Consortium"
+LICENSE="CMake"
+REVISION="1"
+SOURCE_URI="https://cmake.org/files/v${portVersion%.*}/cmake-$portVersion.tar.gz"
+CHECKSUM_SHA256="6d5de15b6715091df7f5441007425264bdd477809f80333fdf95f846aaff88e4"
+PATCHES="cmake-$portVersion.patchset"
+
+ARCHITECTURES="all !x86_gcc2"
+SECONDARY_ARCHITECTURES="x86"
+
+portVersionCompat="$portVersion compat >= ${portVersion%%.*}"
+
+PROVIDES="
+	cmake$secondaryArchSuffix = $portVersionCompat
+	cmd:ccmake = $portVersionCompat
+	cmd:cmake = $portVersionCompat
+	cmd:cpack = $portVersionCompat
+	cmd:ctest = $portVersionCompat
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libarchive$secondaryArchSuffix
+#	lib:libbz2$secondaryArchSuffix
+#	lib:libcppdap$secondaryArchSuffix
+	lib:libcurl$secondaryArchSuffix
+	lib:libexpat$secondaryArchSuffix
+	lib:libjsoncpp$secondaryArchSuffix
+	lib:libform$secondaryArchSuffix	# from ncurses.
+#	lib:liblzma$secondaryArchSuffix
+	lib:libncurses$secondaryArchSuffix
+	lib:librhash$secondaryArchSuffix
+	lib:libstdc++$secondaryArchSuffix
+	lib:libuv$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+#	lib:libzstd$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libarchive$secondaryArchSuffix
+#	devel:libbz2$secondaryArchSuffix	# not actually used even with --system-libs?
+#	devel:libcppdap$secondaryArchSuffix # use the embedded copy to avoid circular deps.
+	devel:libcurl$secondaryArchSuffix
+	devel:libexpat$secondaryArchSuffix
+	devel:libjsoncpp$secondaryArchSuffix >= 25
+#	devel:liblzma$secondaryArchSuffix	# not actually used even with --system-libs?
+	devel:libncurses$secondaryArchSuffix
+	devel:librhash$secondaryArchSuffix
+	devel:libuv$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
+#	devel:libzstd$secondaryArchSuffix	# not actually used even with --system-libs?
+	devel:nlohmann_json
+	"
+BUILD_PREREQUIRES="
+	cmd:gcc$secondaryArchSuffix
+	cmd:grep
+	cmd:ld$secondaryArchSuffix
+	cmd:libtoolize$secondaryArchSuffix
+	cmd:make
+	cmd:sed
+	"
+
+defineDebugInfoPackage cmake$secondaryArchSuffix \
+	$prefix/bin/ccmake \
+	$prefix/bin/cmake \
+	$prefix/bin/cpack \
+	$prefix/bin/ctest
+
+
+BUILD()
+{
+	export LDFLAGS="-lbsd"
+
+	# not an autotools configure
+	./configure --prefix=$prefix \
+		--datadir=/$relativeDataDir/cmake \
+		--docdir=/$relativeDocDir \
+		--mandir=/$relativeManDir \
+		--xdgdatadir=/$relativeDataDir \
+		--system-libs \
+		--no-system-cppdap \
```

Comment:
This is apparently broken. It still tried to use the system `cppdap` and failed, not finding it.

Reply:
>  0x0000000000000001 (NEEDED)             Shared library: [libcppdap.so]

Doesn't that mean that it links the system version? Also note that this is the command line version, not the GUI.

Reply:
I didn't even try to build cmake_gui yet.